spec/operators/sampleTime-spec.ts

```diff
@@ -1,114 +1,150 @@
-      sampleTime(110, rxTestScheduler),
-      mergeMap((x: string) => of(x))
-    );
-
-    expectObservable(result, unsub).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    rxTest.run(({ hot, expectObservable, expectSubscriptions, time }) => {
+      const e1 = hot('----a-^--b----c----d----e----f----|');
+      const e1subs = '      ^---------------!            ';
+      // period             -----------!----------!---------
+      const period = time(' -----------|                 ');
+      const expected = '    -----------c-----            ';
+      const unsub = '       ----------------!            ';
+
+      const result = e1.pipe(
+        mergeMap((x: string) => of(x)),
+        sampleTime(period, rxTest),
+        mergeMap((x: string) => of(x))
+      );
+
+      expectObservable(result, unsub).toBe(expected);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    });
   });
 
   it('should completes if source does not emits', () => {
-    const e1 =  cold('|');
-    const e1subs =   '(^!)';
-    const expected = '|';
-
-    expectObservable(e1.pipe(sampleTime(60, rxTestScheduler))).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    rxTest.run(({ cold, expectObservable, expectSubscriptions, time }) => {
+      const e1 = cold('    |     ');
+      const e1subs = '     (^!)  ';
+      const expected = '   |     ';
+      const period = time('-----|');
+
+      expectObservable(e1.pipe(sampleTime(period, rxTest))).toBe(expected);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    });
   });
 
   it('should raise error if source throws immediately', () => {
-    const e1 =  cold('#');
-    const e1subs =   '(^!)';
-    const expected = '#';
-
-    expectObservable(e1.pipe(sampleTime(60, rxTestScheduler))).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    rxTest.run(({ cold, expectObservable, expectSubscriptions, time }) => {
+      const e1 = cold('    #     ');
+      const e1subs = '     (^!)  ';
+      const expected = '   #     ';
+      const period = time('-----|');
+
+      expectObservable(e1.pipe(sampleTime(period, rxTest))).toBe(expected);
+      expectSubscriptions(e1.subscriptions).toBe(e1subs);
+    });
   });
 
-  it('should not completes if source does not complete', () => {
-    const e1 =  cold('-');
-    const e1subs =   '^';
-    const expected = '-';
-
-    expectObservable(e1.pipe(sampleTime(60, rxTestScheduler))).toBe(expected);
-    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+  it('should not complete if source does not complete', () => {
+    rxTest.run(({ cold, expectObservable, expectSubscriptions, time }) => {
+      const e1 = cold('    --------');
+      const e1subs = '     ^------!';
+      const expected = '   --------';
+      const period = time('-----|  ');
+      const e1unsbs = '    -------!';
+
+      expectObservable(e1.pipe(sampleTime(period, rxTest)), e1unsbs).toBe(expected);
```

Comment:
This is where an explicit unsubscription was added.

Reply:
Makes sense.